torch/_inductor/triton_heuristics.py

```diff
@@ -829,27 +836,52 @@ def cached_autotune(
     save_cache_hook: Optional[Callable[[Any, Any], Any]]
     inductor_meta = {} if inductor_meta is None else inductor_meta
 
-    # on disk caching logic
+    # on disk caching logic and/or remote caching
     if filename is not None and (len(configs) > 1 or config.coordinate_descent_tuning):
         cache_filename = os.path.splitext(filename)[0] + ".best_config"
         configs_hash = hash_configs(configs)
-        best_config = load_cached_autotuning(cache_filename, configs_hash, configs)
+        remote_cache = None
+        remote_cache_key = None
+        if config.use_autotune_remote_cache:
+            from triton.compiler.compiler import make_backend, triton_key
+            from triton.runtime.driver import driver
+
+            target = driver.active.get_current_target()
+            backend = make_backend(target)
+            key = f"{triton_key()}-{backend.hash()}-autotune-best-config"
+            key = hashlib.sha256(key.encode("utf-8")).hexdigest()
+            if config.is_fbcode():
+                from triton.runtime.fb_memcache import FbMemcacheRemoteCacheBackend
+
+                remote_cache = FbMemcacheRemoteCacheBackend(key, is_autotune=True)
+            else:
+                from triton.runtime.cache import RedisRemoteCacheBackend
+
+                remote_cache = RedisRemoteCacheBackend(key)
+            # we already SHA-1 hash the source contents
+            remote_cache_key = os.path.basename(filename)
+
+        best_config = load_cached_autotuning(
+            cache_filename, remote_cache, remote_cache_key, configs_hash, configs
+        )
```

Comment:
nit: you can simplify this a bit if you move remote cache / file operations outside of the `load_cached_autotuning` method
```suggestion
            remote_cache_key = os.path.basename(filename)
            candidate_config = remote_cache.get([remote_cache_key])
       else:
           candidate_config = None
           if os.path.exists(cache_filename):
              with open(cache_filename) as fd:
              candidate_config = json.loads(fd.read())

        best_config = load_cached_autotuning(candidate_config, configs_hash, configs)
```

Reply:
I'd still like to use the cache on disk when it is available, so this would be a behavior change. Having that said this is a function that only gets called from a single place, so unless we plan on calling it from other places, not use how valueable the refactor is (while maintaining the local disk priority)